src/lib/y2network/widgets/wireless_networks.rb

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) [2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "cwm/table"
+
+module Y2Network
+  module Widgets
+    # This class
+    class WirelessNetworks < CWM::Table
+      attr_reader :selected
+
+      # Constructor
+      #
+      # @param networks [Array<WirelessNetwork>] List of available wifi networks
+      def initialize(networks = [])
+        textdomain "network"
+        @networks = networks
+      end
+
+      # Returns table headers
+      #
+      # @return [Array<String>]
+      def header
+        [_("SSID"), _("Mode"), _("Channel"), _("Rate"), _("Signal"), _("Security")]
+      end
+
+      # Returns table items
+      #
+      # Each item corresponds to a wireless network.
+      #
+      # @return [Array<Array<String>>]
+      def items
+        @networks.map do |network|
+          [
+            network.essid,
+            network.essid,
+            network.mode,
+            network.channel,
+            network.rates.max.to_s,
+            network.quality_percent ? "#{network.quality_percent}%" : "",
+            network.auth_mode&.to_human_string
```

Comment:
I would prefer to have the supported security options instead of the auth mode, but we could do it later.

Reply:
Yes, you are right. We need to "rework" the security stuff, but let's do it as part of a new PBI. Somehow, we have been merging several concepts into a single one (authentication mechanism, authentication suites, ciphers, and so on).